xapian-core/queryparser/termgenerator_internal.cc

```diff
@@ -229,9 +270,23 @@ parse_terms(Utf8Iterator itor, unsigned cjk_flags, bool with_positions,
 	    do {
 		Unicode::append_utf8(term, ch);
 		prevch = ch;
-		if (++itor == Utf8Iterator() ||
-		    (cjk_flags && CJK::codepoint_is_cjk(*itor)))
+		if (++itor == Utf8Iterator())
 		    goto endofterm;
+		else if (cjk_flags) {
+		    // Only deal mixed Chinese numbers which start
+		    // with an Arabic digit and next is a Chinese digit.
+		    if (is_digit(prevch) && is_Chinese_digit(*itor)) {
+			do {
+			    ch = *itor;
+			    Unicode::append_utf8(term, ch);
+			} while (++itor != Utf8Iterator() &&
+			(is_digit(*itor) || is_Chinese_digit(*itor)));
```

Comment:
This line is under-indented - the first `(` should line up under the first `+` of the line above.